osquery/extension.go

```diff
@@ -397,10 +397,13 @@ func (e *Extension) writeBufferedLogsForType(typ logger.LogType) error {
 	if err != nil {
 		return err
 	}
-	err = e.db.Update(func(tx *bolt.Tx) error {
+
+	// Collect up logs to be sent
+	logs := []string{}
```

Comment:
I would prefer this starts with `var logs []string`  and takes advantage of Go's ability to append to a declared slice.